### PR TITLE
[SAIC-813] Add timeout and abort_on_prompts options to the RemoteRunner

### PR DESCRIPTION
**Pull request by SVilgelm**
Sometime remote runner cannot raise an exception because
remote host waits an additional information.
Example: ssh server waits login and password when we are
connecting by using wrong key pair.
To solve this problem the timeout option has been added
to the RemoteRunner and abort_on_prompts option has been
set as True.
Remove raising AbortMigration error in case of any hosts
are unreacheable.

### CODE REVIEW

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by archyufa** [comment]
Can we add Warning at the end of the ssh check stage with nodes which has been skipped with timeout.
So they can be looked as the nodes with potential ssh or key issues.

File: cloudferrylib/utils/remote_runner.py [hunk 1]
```diff
@@ -29,7 +29,7 @@ class RemoteExecutionError(RuntimeError):
 
 class RemoteRunner(object):
     def __init__(self, host, user, password=None, sudo=False, key=None,
-                 ignore_errors=False):
+                 ignore_errors=False, timeoute=300):
```
Comment (roman-verchikov) [inline cloudferrylib/utils/remote_runner.py]: `timeout` not `timeoute`

File: cloudferrylib/utils/remote_runner.py [hunk 2]
```diff
@@ -38,6 +38,7 @@ def __init__(self, host, user, password=None, sudo=False, key=None,
         self.sudo = sudo
         self.key = key
         self.ignore_errors = ignore_errors
+        self.timeout = timeoute
```
Comment (roman-verchikov) [inline cloudferrylib/utils/remote_runner.py]: same here

File: cloudferrylib/utils/remote_runner.py [hunk 3]
```diff
@@ -61,9 +62,9 @@ def run(self, cmd, **kwargs):
                 LOG.debug("running '%s' on '%s' host as user '%s'",
                           cmd, self.host, self.user)
                 if self.sudo and self.user != 'root':
-                    return fab_sudo(cmd)
+                    return fab_sudo(cmd, timeout=self.timeout)
                 else:
-                    return run(cmd)
+                    return run(cmd, timeout=self.timeout)
```
Comment (roman-verchikov) [inline cloudferrylib/utils/remote_runner.py]: would lead to `KeyError` because of issues above

File: cloudferrylib/utils/remote_runner.py (hunk 1 above)
Comment (SVilgelm) [inline cloudferrylib/utils/remote_runner.py]: Done

File: cloudferrylib/utils/remote_runner.py (hunk 2 above)
Comment (SVilgelm) [inline cloudferrylib/utils/remote_runner.py]: Done

File: cloudferrylib/utils/remote_runner.py (hunk 3 above)
Comment (SVilgelm) [inline cloudferrylib/utils/remote_runner.py]: No

**Comment by SVilgelm** [comment]
@archyufa check_ssh raises AbortMigrationError in case of any errors and writes error message.

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.os.actions.check_ssh
cloudferrylib/os/actions/check_ssh.py:39: [W0141(bad-builtin), CheckSSH.run] Used builtin function 'map'. Using a list comprehension can be clearer.
cloudferrylib/os/actions/check_ssh.py:40: [W0141(bad-builtin), CheckSSH.run] Used builtin function 'filter'. Using a list comprehension can be clearer.
cloudferrylib/os/actions/check_ssh.py:40: [W0110(deprecated-lambda), CheckSSH.run] map/filter on lambda could be replaced by comprehension
cloudferrylib/os/actions/check_ssh.py:28: [W0613(unused-argument), CheckSSH.run] Unused argument 'info'
cloudferrylib/os/actions/check_ssh.py:60: [W0703(broad-except), CheckSSH.check_access] Catching too general exception Exception

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.os.actions.check_ssh
cloudferrylib/os/actions/check_ssh.py:27: [W0613(unused-argument), CheckSSH.run] Unused argument 'info'
cloudferrylib/os/actions/check_ssh.py:57: [W0703(broad-except), CheckSSH.check_access] Catching too general exception Exception

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.os.actions.check_ssh
cloudferrylib/os/actions/check_ssh.py:27: [W0613(unused-argument), CheckSSH.run] Unused argument 'info'
cloudferrylib/os/actions/check_ssh.py:57: [W0703(broad-except), CheckSSH.check_access] Catching too general exception Exception

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests failed
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-unittests-and-style-checks/1999/

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.os.actions.check_ssh
cloudferrylib/os/actions/check_ssh.py:27: [W0613(unused-argument), CheckSSH.run] Unused argument 'info'

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests failed
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-unittests-and-style-checks/2000/

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.os.actions.check_ssh
cloudferrylib/os/actions/check_ssh.py:27: [W0613(unused-argument), CheckSSH.run] Unused argument 'info'

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.os.actions.check_ssh
cloudferrylib/os/actions/check_ssh.py:26: [W0613(unused-argument), CheckSSH.run] Unused argument 'info'

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests failed
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-unittests-and-style-checks/2001/

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.os.actions.check_ssh
cloudferrylib/os/actions/check_ssh.py:26: [W0613(unused-argument), CheckSSH.run] Unused argument 'info'
************* Module tests.cloudferrylib.utils.test_runner
tests/cloudferrylib/utils/test_runner.py:41: [W0703(broad-except), RemoteRunnerTestCase.test_errors_are_suppressed_for_run_ignoring_errors] Catching too general exception Exception

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.os.actions.check_ssh
cloudferrylib/os/actions/check_ssh.py:26: [W0613(unused-argument), CheckSSH.run] Unused argument 'info'
************* Module tests.cloudferrylib.utils.test_runner
tests/cloudferrylib/utils/test_runner.py:41: [W0703(broad-except), RemoteRunnerTestCase.test_errors_are_suppressed_for_run_ignoring_errors] Catching too general exception Exception

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.os.actions.check_ssh
cloudferrylib/os/actions/check_ssh.py:26: [W0613(unused-argument), CheckSSH.run] Unused argument 'info'
************* Module tests.cloudferrylib.utils.test_runner
tests/cloudferrylib/utils/test_runner.py:41: [W0703(broad-except), RemoteRunnerTestCase.test_errors_are_suppressed_for_run_ignoring_errors] Catching too general exception Exception

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.os.actions.check_ssh
cloudferrylib/os/actions/check_ssh.py:26: [W0613(unused-argument), CheckSSH.run] Unused argument 'info'
************* Module tests.cloudferrylib.utils.test_runner
tests/cloudferrylib/utils/test_runner.py:41: [W0703(broad-except), RemoteRunnerTestCase.test_errors_are_suppressed_for_run_ignoring_errors] Catching too general exception Exception

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by roman-verchikov** [comment]
+1, retest this please

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.os.actions.check_ssh
cloudferrylib/os/actions/check_ssh.py:26: [W0613(unused-argument), CheckSSH.run] Unused argument 'info'
************* Module tests.cloudferrylib.utils.test_runner
tests/cloudferrylib/utils/test_runner.py:41: [W0703(broad-except), RemoteRunnerTestCase.test_errors_are_suppressed_for_run_ignoring_errors] Catching too general exception Exception

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Pull request closed without merging**